File: Problem Set 2/readability.c
// implements the Coleman-Liau index formula to show the reading level of text

#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int calculateLevel(const string text);

int main(void)
{
    string text = get_string("Text: ");
    int level = calculateLevel(text);

    if (level > 16)
    {
        printf("Grade 16+\n");
    }

    else if (level < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %d\n", level);
    }
}

int calculateLevel(const string text)
{
    int i = 0;
    int letters = 0;
    int words = 1;
    int sentences = 0;

    while (text[i] != '\0')
    {
        if (isalpha(text[i]))
        {
            letters++;
        }

        else if (text[i] == ' ')
        {
            words++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }

        i++;
    }

    float L = (letters * 100.0) / words;
    float S = (sentences * 100.0) / words;
    return (int) round(0.0588 * L - 0.296 * S - 15.8);
}

File: Problem Set 6 /readability.py
# Asks the user to type in some text
# Outputs the grade level for the text, according to the Coleman-Liau formula

def main():
    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count


def count_words(text):
    return len(text.split())


def count_sentences(text):
    count = 0
    for char in text:
        if char in [".", "!", "?"]:
            count += 1
    return count


if __name__ == "__main__":
    main()
